keanu-project/src/test/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensorTest.java

```diff
@@ -312,4 +312,35 @@ public void canCalculateProductOfVector() {
         assertTrue(vectorA.isVector() && vectorB.isVector());
     }
 
+    @Test
+    public void scalarMinusInPlaceTensorBehavesSameAsMinus() {
```

Comment:
Similar comment to that above really - what if we had say

{1 , 2, 3} - {{1, 2, 3},
                   {4, 5, 6}}

Wouldn't this need reshaping too?